### DISCUSSION TITLE
Use vector (.svg) files for goal icons instead of .pngs

### DISCUSSION THREAD

**Discussion by LucyGwilliamAdmin** [open-sdg/open-sdg#1444] [Ideas]
Using vector files for the goal icons will allow assistive technology users to customise their appearance e.g colour, font so that they're more accessible.

Most of the changes for this will be done in sdg-translations - need to swap files:
https://github.com/open-sdg/sdg-translations/tree/master/www/assets/img/goals

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning meant to add comment here before going off.

Something Ann and I discussed was for now, we could swap pngs to svgs in English translations folder as their readily available. Then in open sdg could have soemthing like if file extenstion is svg use that otherwise use png? and then we can move over to svgs for other languages over time? This will also give us chance to see if they're available somewhere so don't have to spend lots of time creating/converting files.

What are your thoughts?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin A technical challenge is that Open SDG needs to know the file extension before it tells the browser to load the image. Right now it is assuming PNG: https://github.com/open-sdg/jekyll-open-sdg-plugins/blob/master/lib/jekyll-open-sdg-plugins/sdg_variables.rb#L124

One option may be a configuration option like "goal_image_extension" which countries could set to "svg". This way if an implementation is using English only (like the UK) then they could choose to use "svg", while the default of "png" would be used for everyone else. If we did that, the next step would be to get SVG versions uploaded to sdg-translations [here](https://github.com/open-sdg/sdg-translations/tree/master/www/assets/img/goals).

I also just thought about the question of high-contrast. Technically if we're using SVG then we wouldn't need a separate image for high-contrast. But Open SDG is coded to use different images. So there may be some tweaks necessary to avoid doing that if SVG is being used.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning  That sounds like a good alternative - I will upload the English svg images before I finish today. 

How much of a tweak will it need if svg images are used? If easier for now, I can upload same images for high contrast versions for now?

**Comment by LucyGwilliamAdmin** [comment]
I should leave the English pngs there aswell right?

**Comment by LucyGwilliamAdmin** [comment]
Sorry for all the comments - they do actually have black and white svgs in the download - should I upload those to high contrast folder?

**Comment by brockfanning** [comment]
Ah ok cool. Yes to all - leave the PNGs there, and upload those black and white SVGs to the high contrast folder as well.

**Comment by LucyGwilliamAdmin** [comment]
Have created PR in sdg-translations for this:
https://github.com/open-sdg/sdg-translations/pull/139

**Comment by brockfanning** [comment]
This has been implemented in jekyll-open-sdg-plugins, so it's ready to test. You can point to jekyll-open-sdg-plugins version 1.1.0 in order to test this. The documentation for the setting is part of this PR: #889. Specifically it's here: https://github.com/open-sdg/open-sdg/pull/889/files#diff-76e731333fb756df3bff5ddb3b731c46R330-R332

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I've just tried testing this by pointing at 1.1.0 and svgs are being used instead of pngs. However, when I use Windows High Contrast mode, the colours in the goals grid still aren't changing.

I just did a quick google and found [this](https://www.scottohara.me/blog/2019/02/12/high-contrast-aria-and-images.html), which talks about Windows High Contrast mode not working when images are styled with css - do you think this could be the case here?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin I believe I have a better handle on this now. The SVG file itself has color baked into it. For our custom high-contrast mode, we actually switch to a completely different image (the version in the high-contrast folder) which is only black-and-white. The Windows high-contrast mode knows nothing of this different image, and so continues displaying the original version which has color.

So I don't think there's anything we can do about that. The only possible solution would be to somehow detect in our custom CSS when the user is using Windows high-contrast mode, and react accordingly. [This CSS rule](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/-ms-high-contrast) theoretically does that, but there's a big warning that it may change in the future and will be inconsistent anyway. So I don't think that's an option. Also if we start adding custom code to support Windows high-contrast mode, we would also need to think about other OS-level high-contrast modes. So that may be a slippery slope we want to avoid.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning ahh ok - I think one to raise with DAC then as not really sure how to move forward with it

**Comment by LucyGwilliamAdmin** [comment]
DAC's consultancy response to "Even though we have swapped pngs for svgs, images still aren’t showing as high contrast in Windows High Contrast Mode. Any idea why?"

> Yes, so for them to invert, instead of:
> 
> **Current Code Ref(s):** #goal-1
> 
> ```
> <img src="https://open-sdg.github.io/sdg-translations/assets/img/goals/en/1.svg" id="goal-1" alt="No poverty - Goal 1">
> ```
> 
> You could use:
> 
> ```
> <div class="col-xs-4 col-sm-2 col-md-1 goal-tile" style="background-color:#e5243b;">
>   <a href="/dac2020/1" style="color:#fff;" aria-label="1. No poverty">
>      <svg xmlns="http://www.w3.org/2000/svg" id="Layer_1" data-name="Layer 1" viewBox="0 0 1000 1000" focusable="false"><link xmlns="" type="text/css" rel="stylesheet" id="dark-mode-general-link"></link><link xmlns="" type="text/css" rel="stylesheet" id="dark-mode-custom-link"></link><style xmlns="" type="text/css" id="dark-mode-custom-style"></style>
>  <defs>
>    <style>
>
>      .cls-2 {
>        fill: currentColor;
>      }
>    </style>
>  </defs>
>  <title>1. No poverty</title>
>  
>  <polygon class="cls-2" points="299.13 113.19 299.13 164.56 287.55 164.56 287.55 82.8 301.04 82.8 315.89 129.72 315.89 82.8 327.35 82.8 327.35 164.56 315.29 164.56 299.13 113.19"></polygon>
>  <path class="cls-2" d="M357.12,165.54c14.61,0,20.86-9.41,20.86-22.08V103.91c0-12.8-6.25-22.07-20.86-22.07s-20.74,9.27-20.74,22.07v39.55c0,12.67,6.27,22.08,20.74,22.08m-7-63.81c0-5,2.17-8.43,7-8.43s7.12,3.48,7.12,8.43v43.9c0,5-2.18,8.44-7.12,8.44s-7-3.49-7-8.44Z"></path>
>  <path fill="currentColor" class="cls-2" d="M301.3,257.64h6.62c13.88,0,18.94-7,18.94-18.45V222c0-11.46-5.06-18.34-18.94-18.34H287.55v81.77H301.3Zm0-42.81h5.06c4.71,0,6.75,2.18,6.75,6.63v18.8c0,4.48-2,6.77-6.75,6.77H301.3Z"></path>
>  <path class="cls-2" d="M355.8,286.34c14.61,0,20.87-9.4,20.87-22.06V224.71c0-12.78-6.26-22.06-20.87-22.06s-20.74,9.28-20.74,22.06v39.57c0,12.66,6.27,22.06,20.74,22.06m-7-63.79c0-5,2.16-8.44,7-8.44s7.13,3.49,7.13,8.44v43.89c0,4.95-2.18,8.45-7.13,8.45s-7-3.5-7-8.45Z"></path>
>   <polygon class="cls-2" points="411.75 285.38 427.31 203.61 414.4 203.61 404.75 260.65 404.62 260.65 395.22 203.61 381.23 203.61 396.79 285.38 411.75 285.38"></polygon>
>  <polygon class="cls-2" points="466.98 274.17 447.93 274.17 447.93 249.31 461.55 249.31 461.55 238.22 447.93 238.22 447.93 214.83 466.98 214.83 466.98 203.61 434.18 203.61 434.18 285.38 466.98 285.38 466.98 274.17"></polygon>
>  <path class="cls-2" d="M489.16,247.87l12.9,37.51h14.48l-13.15-37.15c7.84-1.8,11-7,11-15.8V222c0-11.46-5.07-18.34-18.94-18.34h-20v81.77h13.76Zm0-33h4.57c4.71,0,6.76,2.18,6.76,6.63v12c0,4.35-2.05,6.75-6.76,6.75h-4.57Z"></path>
>  <polygon class="cls-2" points="531.48 285.38 545.35 285.38 545.35 214.83 556.69 214.83 556.69 203.61 520.14 203.61 520.14 214.83 531.48 214.83 531.48 285.38"></polygon>
>  <polygon class="cls-2" points="577.29 285.38 591.04 285.38 591.04 254.27 606.23 203.61 593.09 203.61 584.78 234.6 584.66 234.6 576.33 203.61 561.99 203.61 577.29 254.27 577.29 285.38"></polygon>
>  <polygon class="cls-2" points="139.62 131.52 139.92 131.52 139.92 285.86 173.81 285.86 173.81 84.23 146.74 84.23 115.53 120.81 115.53 156.19 115.83 156.19 139.62 131.52"></polygon>
>  <g>
>    <path class="cls-2" d="M770.13,477.09a27.67,27.67,0,1,1,27.7,27.69,27.7,27.7,0,0,1-27.7-27.69"></path>
>    <path class="cls-2" d="M830.15,515.45a36.89,36.89,0,0,1,33.5,36.77V629a10.73,10.73,0,0,1-21.46,0V563.05c0-2.48-.71-5.82-5.23-5.82-3.13,0-3.78,3.31-3.89,5.42V760.19a15.43,15.43,0,0,1-30.86,0V646.36c0-1.26-.46-4.52-4.38-4.52h0c-3.93,0-4.41,3.26-4.45,4.52V760.19a15.42,15.42,0,1,1-30.83,0V562.65c-.1-2.11-.78-5.42-3.88-5.42-4.54,0-5.23,3.34-5.23,5.82V629A10.73,10.73,0,0,1,732,629V552.22a36.91,36.91,0,0,1,33.55-36.77,15.26,15.26,0,0,1,2-.17H828a16.4,16.4,0,0,1,2.11.17"></path>
>  </g>
>  <g>
>    <path class="cls-2" d="M605.5,491.63a26.21,26.21,0,1,0-26.19,26.22,26.21,26.21,0,0,0,26.19-26.22"></path>
 >   <path class="cls-2" d="M526.48,551,504.69,622s-3.88,13,7.15,16.25,14.31-9.44,14.31-9.44l17.57-59.21s1-4.56,4.38-3.57,2.14,5.85,2.14,5.85l-30.58,111H548.1v79.47a13.66,13.66,0,0,0,27.31,0V682.79h7.76v79.47a13.67,13.67,0,0,0,27.33,0V682.79h28.43l-30.56-111S607.08,567,610.5,566s4.38,3.57,4.38,3.57l17.57,59.21s3.34,12.69,14.3,9.44S653.9,622,653.9,622L632.11,551s-7.78-20.51-25-20.51H551.53c-17.24,0-25,20.51-25,20.51"></path>
>  </g>
>  <g>
>    <path class="cls-2" d="M448.49,491.63a26.21,26.21,0,1,0-26.2,26.22,26.22,26.22,0,0,0,26.2-26.22"></path>
>    <path class="cls-2" d="M369.46,551,347.68,622s-3.9,13,7.14,16.25,14.31-9.44,14.31-9.44l17.57-59.21s.95-4.56,4.38-3.57,2.13,5.85,2.13,5.85l-30.57,111h28.44v79.47a13.66,13.66,0,0,0,27.32,0V682.79h7.76v79.47a13.66,13.66,0,0,0,27.31,0V682.79h28.45l-30.56-111s-1.3-4.87,2.11-5.85,4.39,3.57,4.39,3.57l17.56,59.21s3.34,12.69,14.31,9.44S496.87,622,496.87,622L475.08,551s-7.77-20.51-25-20.51H394.51c-17.23,0-25.05,20.51-25.05,20.51"></path>
>  </g>
>  <g>
>    <path class="cls-2" d="M693.54,638.35a21.82,21.82,0,1,0-21.82-21.84,21.8,21.8,0,0,0,21.82,21.84"></path>
>    <path class="cls-2" d="M641.23,706.13c-1.07,2.55-1.94,8,3.18,10.32,6,2.76,9.35-3,9.35-3l11.86-37.83s2.06-2.53,2.06.09v89.05h.05a11.2,11.2,0,0,0,22.39,0V729.57s-.65-5,3.36-5,3.38,5,3.38,5v35.15a11.22,11.22,0,0,0,22.44,0V675.67c0-2.62,1.38-.75,1.38-.75l9.3,25.9s4,11.57,5.28,12.08a8.26,8.26,0,0,0,10.81-4.47,8.16,8.16,0,0,0-.88-8l.08-.08c-.27-.84-13.9-34.26-15.5-38.95-4.92-14.37-13.88-14.24-22.5-14.48s-13.79,0-13.79,0-5.51-.22-16.9.11-15,6.19-27.51,38.46c-1.79,4.65-7.84,19.68-7.92,20.58Z"></path>
>  </g>
>  <g>
>    <path class="cls-2" d="M186.66,504.78A27.69,27.69,0,1,0,159,477.09,27.72,27.72,0,0,0,186.66,504.78Z"></path>
>    <path class="cls-2" d="M219.22,515.31A18.06,18.06,0,0,0,217,515h-60.5a13.66,13.66,0,0,0-2,.31A37,37,0,0,0,121,552.22V629a10.66,10.66,0,0,0,5.61,9.46,4.07,4.07,0,0,0-.36.94l-24.6,130.3a4,4,0,0,0,3.22,4.74,3.81,3.81,0,0,0,.75.07,4.05,4.05,0,0,0,4-3.31l24.58-130.3a4,4,0,0,0-.15-1.51A10.62,10.62,0,0,0,142,629V563.05c0-2.48.65-5.82,5.17-5.82,3.12,0,3.83,3.31,3.83,5.42V760.19a15.5,15.5,0,0,0,31,0V646.36a4.3,4.3,0,0,1,4.66-4.36h0c3.93,0,4.33,3.1,4.33,4.36V760.19a15.5,15.5,0,0,0,31,0V562.65c0-2.11.74-5.42,3.83-5.42,4.55,0,5.17,3.34,5.17,5.82V629a11,11,0,0,0,22,0V552.22A37.25,37.25,0,0,0,219.22,515.31Z"></path>
>  </g>
>  <g>
>    <path class="cls-2" d="M326.16,674.41l20.25-36.75s2.44-5.28,2.44-6.54a8,8,0,0,0-14.72-4.21l-.09,0c-.64.55-10.83,18.63-14.19,22-3.55,3.59-6.74,3.14-6.74,3.14H286.67s-3.2.45-6.74-3.14c-3.4-3.36-13.54-21.47-14.19-22l-.07.06A8,8,0,0,0,251,631.11a14.41,14.41,0,0,0,.78,3.49l21.95,39.81a14.62,14.62,0,0,1,1.29,5.25v7.84L263.5,730H275v35h.15a10.94,10.94,0,0,0,10.94,10.83A10.81,10.81,0,0,0,297,765.14V731.41a1.8,1.8,0,0,1-.38-1.41h6.5a3.39,3.39,0,0,1-.12,1.41v33.73a11,11,0,0,0,22,0V730h11.29L325,687.25v-7.59A15.92,15.92,0,0,1,326.16,674.41Z"></path>
>    <path class="cls-2" d="M299.88,638.35a21.82,21.82,0,1,0-21.81-21.84A21.8,21.8,0,0,0,299.88,638.35Z"></path>
>  </g>
> </svg>
> </a>
> </div>
> ```

>(please note I deleted `<rect xmlns="http://www.w3.org/2000/svg" class="cls-1" width="1000" height="1000"/>`  so I would specify the width and height of the div instead)

> This ensures that the text turns yellow in high contrast mode (as it is linktext) and that the background colour disappears.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning what are your thoughts on DAC's response?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin The big change there is adding the SVG "inline" in the HTML of the page, rather than referring to it in an image's "src" attribute. We'll need to think more on how that could work. I think it's safe to say that it will be much more of a revamp than our `goal_image_extension` solution was.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning what are your thoughts on how long this might take?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I removed the colour from the svgs, so they now look like this:
https://lucygwilliamadmin.github.io/sdg-indicators/

I think next step might be adding colour using css? I did take a brief look but abilities don't stretch that far 👎 

1: `#e5243b`
2: `#DDA63A`
3. `#4C9F38`
4. `#C5192D`
5. `#FF3A21`
6. `#26BDE2`
7. `#FCC30B`
8. `#A21942`
9. `#FD6925`
10. `#DD1367`
11. `#FD9D24`
12. `#BF8B2E`
13. `#3F7E44`
14. `#0A97D9`
15. `#56C02B`
16. `#00689D`
17. `#19486A`

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin That approach seems promising. You can see that color is being added on hover - maybe you could apply that same approach all the time. (Offhand, I'm not sure why we're doing that on-hover thing.)

That code is here: https://github.com/open-sdg/open-sdg/blob/master/_sass/components/_goal_tiles.scss#L16-L20

Maybe it could just be _always_ applied, like so?

```
.goal-tiles #goal-#{$i} {
  background-color: $c;
}
```

Side note: I'm interested to see if this would mean that we don't need different images for high-contrast. In theory could it just be one image which is white-on-transparent, and we set its background color as the goal color normally, but black in high-contrast?

**Comment by LucyGwilliamAdmin** [comment]
Tried that here http://sdgdev-813006012.eu-west-1.elb.amazonaws.com/dac2020-svgs/

Only applies to frontpage and goal page though?

I think that's right re. high contrast.

I think the hover thing might have been implemented when the drop shadow was added - I might be wrong though

**Comment by LucyGwilliamAdmin** [comment]
Think `goal-tiles` class needs to be added divs that contain images and `id=goal-#` needs to be added to the image elements:
![image](https://user-images.githubusercontent.com/43404358/94288112-f6b7e400-ff4e-11ea-9637-ef8d2a6ce118.png)

Tried in inspect panel for goal 1 page

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin I've approved your PRs - from what I can tell this approach works great in Windows high-contrast, and did not require the inline SVG code that DAC recommended. Nice!

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin Do you think we should close this and start new issues, or continue in this one? As I understand it, next steps might be:
1. Figure out how to use the same SVG image in our high-contrast mode rather than loading a separate high-contrast version of the image
2. Add the non-English SVGs to sdg-translations.

Those should be done in order, as our solution for the first item may affect what folder they need to be in, on sdg-translations.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning  I think it might be good to start a new ticket - I'm happy to create one if you agree?

**Comment by brockfanning** [comment]
Let's contact UNSD about whether they have some source files that would make it easier to convert PNGs to SVGs.

**Comment by phillipgardiner** [comment]
@LucyGwilliamAdmin and @brockfanning Not entirely sure what you are looking for, but just making sure you were aware of this page: https://www.globalgoals.org/resources. It seems to have vectors available and provide the icons in 20 different languages.

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin I opened #1039 as a first step towards better SVG support. This should prevent us from needing high-contrast versions of the SVGs.

**Comment by LucyGwilliamAdmin** [comment]
Removing 1.3.0 label from this and not changing to 1.5.0 as it's dependent on SVGs being available - I have queried UN but no response
